### PR TITLE
Stable move to actions

### PR DESCRIPTION
**Pull request by snoyberg**
Moves unit tests (aka not the nightly tests) from Azure CI over to Github Actions.

### CODE REVIEW

**Comment by snoyberg** [comment]
@psibi would you mind reviewing this? The integration tests (nightly) isn't yet converted, so it's still failing, but I figured it would be good to do the first step of conversion first.

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: hlint
+        run: |
+          set -ex
+          ./etc/scripts/get-hlint.sh
+          export PATH="$(pwd)"/hlint:$PATH
+          hlint src/
+          hlint src/ --cpp-define=WINDOWS=1
+          hlint test/ --cpp-simple
```
Comment (psibi) [inline .github/workflows/unit-tests.yml:21]: I guess we could use the one available in marketplace: https://github.com/marketplace/actions/haskell-linter

But it's unofficial (not from github) and we have a better control here. From looking at the repo, I wasn't able to find which hlint version it was using.

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          extra-suffix: ""
+          stack-args: ""
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-${{ hashFiles('${{ matrix.stack-yaml }}') }}-${{ matrix.extra-suffix }}
```
Comment (psibi) [inline .github/workflows/unit-tests.yml:75]: Do you think `package.yaml` should be part of the cache hash ?

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (psibi) [reply .github/workflows/unit-tests.yml:21]: I guess adding hlint to the official one would add value: https://github.com/github/super-linter

Comment (snoyberg) [reply .github/workflows/unit-tests.yml:21]: No objection to doing that in the future, but I don't want to block this PR on it. I also don't want to rely on something unofficial, especially when the solution we have has worked on Azure CI for years.

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (snoyberg) [reply .github/workflows/unit-tests.yml:75]: I could go both ways, but I lean towards no. Basically: if package.yaml changes, but stack.yaml doesn't, it means that packages were added or removed, but the package set itself is unchanged. That means that the cache shouldn't be invalidated.

**Pull request merged**